### PR TITLE
[Example] add sdf2d.py

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #... (if any)

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

**Comment by k-ye** [comment]
Could you attach a screenshot?

**Comment by archibate** [comment]
Of course!
Screenshot_20200421_215408

File: examples/sdf2d.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    return a
+
+@ti.func
+def intersect(a, b):
+    if a[0] < b[0]:
+        a = b
+    return a
+
+@ti.func
+def reflect(r, n):
```
Comment (k-ye) [inline examples/sdf2d.py:34]: FYI I just added this https://github.com/taichi-dev/taichi/blob/5db9f22074f8da749edf8869b8854390fe117f75/examples/renderer_utils.py#L23-L25

Comment (archibate) [reply examples/sdf2d.py:34]: Thanks for the useful stuff! Btw #838 includes some of these too.

**Comment by archibate** [comment]
> https://zhuanlan.zhihu.com/p/30816284

Yeah that's where I start from. I'm considering to make it colorful later.

**Comment by yuanming-hu** [comment]
If you guys haven't seen this, this post is also worth checking out:
https://benedikt-bitterli.me/tantalum/
https://benedikt-bitterli.me/tantalum/tantalum.html

**Comment by k-ye** [comment]
Thanks!

**Pull request merged**